vault/core.go

```diff
@@ -2632,6 +2651,53 @@ func (c *Core) SetConfig(conf *server.Config) {
 	c.logger.Debug("set config", "sanitized config", string(bz))
 }
 
+func (c *Core) SetCustomResponseHeaders(w http.ResponseWriter, status int) {
+	if c.customListenerHeader == nil {
+		c.logger.Debug("failed to find the custom response headers configuration")
+		return
+	}
+
+	c.customListenerHeader.SetCustomResponseHeaders(w, status)
+}
+
+func (c *Core) ExistCustomResponseHeader(header string, la string) bool {
+	if c.customListenerHeader == nil {
+		c.logger.Debug("failed to find the custom response headers configuration")
+		return false
+	}
+
+	return c.customListenerHeader.ExistCustomResponseHeader(header, la)
+}
+
+func (c *Core) ReloadCustomResponseHeaders() error {
+
+	conf := c.rawConfig.Load()
+	if conf == nil {
+		return fmt.Errorf("failed to load core raw config")
+	}
+
+	tempLH := c.customListenerHeader
+	c.customListenerHeader = nil
+
+	uiHeaders, _ := c.UIHeaders()
+
+	customHeaderLogger := c.GetLogger("customHeader")
```

Comment:
We usually only create a new sub logger for new subsystems.  I'm not sure what the dividing line should be, but e.g. if there's a new background goroutine, I could see that warranting a new logger.  I don't think this feature warrants one.

Reply:
So, the reason I did this is that we need to log invalid headers like the ones start with "X-Vault" and the conflicting ones with the /ui endpoint. So, should I just pass in core.logger? There are other warnings/errors which are logged. Not sure if all is needed though.